test/e2e/framework/console-operator.go

```diff
@@ -0,0 +1,390 @@
+		}, {
+			name:           "Progressing should be False",
+			conditionType:  operatorsv1.OperatorStatusTypeProgressing,
+			expectedStatus: operatorsv1.ConditionFalse,
+			mustBeSet:      true,
+		}, {
+			name:           "Degraded should be False",
+			conditionType:  operatorsv1.OperatorStatusTypeDegraded,
+			expectedStatus: operatorsv1.ConditionFalse,
+			mustBeSet:      true,
+		},
+	}
+
+	for _, tt := range tests {
+		// if the condition is essential, we test mustBeSet. future conditions may be optional
+		// as we are working on transitioning
+		if _, ok := conditions[tt.conditionType]; !ok {
+			if tt.mustBeSet {
+				unmetConditions = append(unmetConditions, tt.conditionType)
+				settled = false
+			}
+		}
+		if conditions[tt.conditionType].Status != tt.expectedStatus {
+			unmetConditions = append(unmetConditions, tt.conditionType)
+			settled = false
+		}
+	}
+	return settled, unmetConditions
+}
+
+func operatorIsObservingCurrentGeneration(operatorConfig *operatorsv1.Console) bool {
+	if operatorConfig.Status.ObservedGeneration != operatorConfig.ObjectMeta.Generation {
+		fmt.Printf("waiting for observed generation %d to match generation %d... \n", operatorConfig.Status.ObservedGeneration, operatorConfig.ObjectMeta.Generation)
+		return false
+	}
+	// fmt.Printf("observed generation %d matches generation %d \n", operatorConfig.Status.ObservedGeneration, operatorConfig.ObjectMeta.Generation)
+	return true
+}
+
+// A helper to ensure our operator config reaches a settled state before we
+// begin the next test.
+func WaitForSettledState(t *testing.T, client *ClientSet) (settled bool, err error) {
+	t.Helper()
+	fmt.Printf("waiting to reach settled state...\n")
+	// don't rush it
+	interval := 2 * time.Second
+	// it should never take this long for a test to pass
+	max := 240 * time.Second
+	count := 0
+	unmetConditions := []string{}
+	pollErr := wait.Poll(interval, max, func() (stop bool, err error) {
+		// lets be informed about tests that take a long time to settle
+		count++
+		logUnsettledAtInterval(count)
+		operatorConfig, err := client.Operator.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		// first, wait until we are observing the correct generation. if we are still looking at a previous
+		// generation, pass on this iteration of the loop
+		isCurrentGen := operatorIsObservingCurrentGeneration(operatorConfig)
+		if !isCurrentGen {
+			return false, nil
+		}
+		// then wait until the operator status settle
+		settled, unmet := operatorIsSettled(operatorConfig)
+		unmetConditions = unmet // avoid shadow
+		return settled, nil
+	})
+	if pollErr != nil {
+		t.Errorf("operator has not reached settled state in %v attempts due to %v - %v", max, unmetConditions, pollErr)
+	}
+	return true, nil
+
+}
+
+// short term helper to simply print how long it takes to reach a settled state at certain intervals
+func logUnsettledAtInterval(count int) {
```

Comment:
👍

Reply:
Just to give us clues about longer running tests.

Reply:
It does add extra logging noise, but if we have a flaky test again, it's pretty helpful for diagnosing.